integration/aggregate_compat_test.go

```diff
@@ -77,8 +82,9 @@ func testAggregateStagesCompatWithProviders(t *testing.T, providers shareddata.P
 				}
 			}
 
-			if !hasSortStage {
-				// add sort stage to sort by _id because compat does not have deterministic order.
+			if !hasSortStage && !tc.noSort {
+				// add sort stage to sort by _id because compat and target
+				// would be ordered differently otherwise.
 				pipeline = append(pipeline, bson.D{{"$sort", bson.D{{"_id", 1}}}})
```

Comment:
To be honest, I'm not a fan of `noSort` variable in tests. Maybe we should just explicitly set sort in each test where it's needed and remove this line completely? Then we don't need `hasSortStage` nor `noSort`?

Reply:
Removed noSort in preference for just checking empty pipeline